docassemble/AssemblyLine/al_document.py

```diff
@@ -1445,8 +1445,19 @@ def send_email(
                 **kwargs,
             )
 
-    def is_enabled(self, refresh=True):
-        return self.has_enabled_documents(refresh=refresh)
+    def is_enabled(self, refresh=True) -> bool:
+        """Returns true if the bundle itself is enabled, and it has at least one enabled child document"""
+        self_enabled = (hasattr(self, "always_enabled") and self.always_enabled) or \
+            (hasattr(self.cache, "enabled") and self.cache.enabled)
+        if not self_enabled:
+            if refresh:
+                self.cache.enabled = self.enabled
```

Comment:
How does this work if:
- always_enabled is False
- the user changes an option that makes the bundle NOT enabled?

Reply:
I should have moved the second half of the `or` statement inside `if not self_enabled`, but either way, if `always_enabled is False` nothing will turn it on.

And yeah, on refresh it should reconsidered everything but `always_enabled` (if it's not in the cache).

The logic was just taken from `ALDocument.is_enabled`. It's not in the same object hierarchy, so I can't just call it.

Reply:
I think `always_enabled` is only supposed to be True if `enabled` is set to true inside the `init` method, i.e., in the `.using()` method in the `objects` block. So in most of the interesting situations it will start out `False`.

I just am making sure this logic works if `always_enabled` starts out `False`, someone enables it (so the .cache.enabled property is at least temporarily true) and then they change their mind.

The answer might be that the `cache` is the only place we're checking and that's erased on screen refresh. If so that's good--and I think it's working correctly in `ALDocument.is_enabled()`

Reply:
Misunderstood how always_enabled was supposed to work. The above should just be the same ALDocument, I'm not trying to make it any different, but refactored it a bit to make that clearer.